File: 0001 Code Chef/0018 Saving Taxes/implementation/0010 nodejs/saving_taxes.js
const readline = require('readline');

const rl = readline.createInterface({
    input:process.stdin,
    output:process.stdout
});

let count = 0;


rl.question('',(input)=>{
    let test = parseInt(input);
    rl.on('line',(input) => {
        let string_array = input.split(" ");
        let x = parseInt(string_array[0]);
        let y = parseInt(string_array[1]);
        if(x > y){
            console.log(x-y);
        }else{
            console.log(0);
        }
        count = count + 1;
        if(count == test){
            rl.close();
        }

    });
});

File: 0001 Code Chef/0018 Saving Taxes/implementation/0005 kotlin/SavingTaxes.kt
fun main(args:Array<String>){
    var stringInput:String? = readLine();
    if(stringInput != null){
        var test:Int? = stringInput.toIntOrNull();
        if(test != null){
            var count:Int = test;
            while(count > 0){
                var stringInput:String? = readLine();
                if(stringInput != null){
                    var stringArray:Array<String> = stringInput.split(" ").toTypedArray();
                    if(stringArray.size == 2){
                        var x:Int? = stringArray[0].toIntOrNull();
                        var y:Int? = stringArray[1].toIntOrNull();
                        if(x != null && y != null){
                            if(x > y){
                                var savings:Int = x - y;
                                println(savings);
                            }else{
                                println(0);
                            }
                        }
                    }
                }
                count = count - 1;
            }
        }
    }
}